qemu/tests/cfg/virtio_driver_sign_check.cfg

```diff
@@ -1,16 +1,32 @@
 - virtio_driver_sign_check:
     virt_test_type = qemu
-    only Win7.x86_64
-    # This script only need run once for every virtio driver version.
-    # So need not add to our testing loop.
+    only Windows
     type = virtio_driver_sign_check
-    cdroms += " sdk virtio"
-    cdrom_virtio = isos/windows/virtio-win.iso
-    cdrom_sdk = isos/windows/GRMSDKX_EN_DVD.iso
+    cdroms += " virtio"
+    cdrom_virtio = isos/windows/virtio-win.latest_prewhql.iso
+    floppy_readonly = yes
     floppies = "fl"
-    floppy_name = images/win7-64/virtio-win.vfd
-    list_files_cmd = dir /s /b %s | find "%s"
-    winsdk_au3 = autoit/winsdk.au3
-    signtool_install = "D:\autoit3.exe c:\winsdk.au3"
-    signtool_cmd = "c:\Program Files\Microsoft SDKs\Windows\v7.0\Bin\signtool.exe" verify /v /kp /c %s %s
-    drive_list = F: A:
+    list_files_cmd = 'dir /s /b %s | find "%s"'
+    get_disk_letter_cmd = 'wmic logicaldisk | find "virtio-win"'
+    drive_volume = A:
+    tested_drivers = Balloon pvpanic viorng vioserial vioinput NetKVM netkvm vioscsi viostor   
+    guest_alias = "Win7-32:w7\x86,Win7-64:w7\amd64,Win8-32:w8\x86,Win8-64:w8\amd64,Win8-32.1:w8.1\x86,Win8-64.1:w8.1\amd64,Win10-32:w10\x86,Win10-64:w10\amd64,Win2008-sp2-32:2k8\x86,Win2008-sp2-64:2k8\amd64,Win2008-r2-64:2k8R2\amd64,Win2012-64:2k12\amd64,Win2012-64r2:2k12R2\amd64,Win2016-64:2k16\amd64"
+    guest_alias_fl = "Win7-32:i386\Win7,Win7-64:amd64\Win7,Win8-32:i386\Win8,Win8-64:amd64\Win8,Win8-32.1:i386\Win8.1,Win8-64.1:amd64\Win8.1,Win10-32:i386\Win10,Win10-64:amd64\Win10,Win2008-sp2-32:i386\Win2008,Win2008-sp2-64:amd64\Win2008,Win2008-r2-64:amd64\Win2008R2,Win2012-64:amd64\Win2012,Win2012-64r2:amd64\Win2012R2,Win2016-64:amd64\Win2016"
+
+    i386:
+        floppy_name = isos/windows/virtio-win.latest_prewhql.vfd.i386
+        Win2008..sp2:
+            signtool = "WIN_UTILS:\signtool\signtool-WLK\Signtool_x86_WLK.exe"
+        Win7, Win8:
+            signtool = "WIN_UTILS:\signtool\signtool-HCK\Signtool_x86_HCK.exe"
+        Win10:
+            signtool = "WIN_UTILS:\signtool\signtool-HLK\Signtool_x86_HLK.exe"
+    x86_64:
+        floppy_name = isos/windows/virtio-win.latest_prewhql.vfd.x86_64
+        Win2008..sp2:
+            signtool = "WIN_UTILS:\signtool\signtool-WLK\Signtool_amd64_WLK.exe"
+        Win2008..r2, Win7, Win8, Win2012:
+            signtool = "WIN_UTILS:\signtool\signtool-HCK\Signtool_amd64_HCK.exe"
+        Win10, Win2016:
+            signtool = "WIN_UTILS:\signtool\signtool-HLK\Signtool_amd64_HLK.exe"
+    signtool_cmd = ${signtool} verify /v /kp /c %s %s
```

Comment:
Is there some iproblem to move this line to line 15 ?

Reply:
yes, the ${signtool} must be used behind signtool definition, if not, will cannot load the correct signtool string~